src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            char currentCharacter;
+            StringBuilder tokenBuffer = null;
+            var outputBuffer = new StringBuilder();
+            var isToken = false;
+            var isEncode = true;
+
+            while (position < length)
+            {
+                currentCharacter = resourceString[position];
+
+                position++;
+                if (currentCharacter == '}')
+                {
+                    if (position < length && resourceString[position] == '}')  // Treat as escape character for }}
+                    {
+                        if (isToken)
+                        {
+                            // for cases like {{0:x}} where we have 2 curly braces, encoding should not happen. 
+                            // Hence setting isEncode to false.
+                            isEncode = false;
+                            tokenBuffer.Append("}}");
+                        }
+                        else
+                        {
+                            outputBuffer.Append("}");
+                        }
+
+                        position++;
+                        if (position == length)
+                        {
+                            AppendToOutputBuffer(isEncode, arguments, tokenBuffer, outputBuffer);
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        AppendToBuffer(isToken, "}", tokenBuffer, outputBuffer);
+
+                        AppendToOutputBuffer(isEncode, arguments, tokenBuffer, outputBuffer);
+
+                        if (position == length)
+                        {
+                            break;
+                        }
+
+                        isToken = false;
+                        tokenBuffer = null;
+                        isEncode = true;
+                    }
+                }
+                else if (currentCharacter == '{')
+                {
+                    if (position < length && resourceString[position] == '{')  // Treat as escape character for {{
+                    {
+                        if (isToken)
+                        {
+                            tokenBuffer.Append("{{");
+                        }
+                        else
+                        {
+                            outputBuffer.Append("{");
+                        }
+                        position++;
+                    }
+                    else
+                    {
+                        tokenBuffer = new StringBuilder();
+                        tokenBuffer.Append("{");
+                        isToken = true;
+                    }
+                }
+                else
+                {
+                    AppendToBuffer(isToken, currentCharacter.ToString(), tokenBuffer, outputBuffer);
```

Comment:
Sorry missed `string` allocation last time. Change `AppendToBuffer()` to `(..., char value, ...)`, pass `'}'` at line 133 and `currentCharacter` here.